### PR TITLE
Enhance category semantics

### PR DESCRIPTION
**Pull request by tschmidtb51**
- addresses parts of https://github.com/oasis-tcs/csaf/issues/426
- add regex to prevent start and end with white space or dash or underscore or dot
- change examples in schema in preparation of namespace
- rephrase introduction in section 4
- rewrite profiles
- change values as suggested
- adopt test 6.1.26
- update relevant tests
- update test files

### CODE REVIEW

**Comment by tschmidtb51** [comment]
@sthagen Currently, we also prohibit the names of the profiles. Did you want to get rid of that as well?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -3798,8 +3817,8 @@ It MUST be tested that at least one item in `/document/notes` exists which has a
 The relevant values for `/document/category` are:
 
 ```
-  security_incident_response
-  informational_advisory
+  csaf_security_incident_response
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:3820]: We might want to sort.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 2]
```diff
@@ -3829,8 +3848,8 @@ It MUST be tested that at least one item in `/document/references` exists that h
...
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:3851]: We might want to sort here too.

**Review by sthagen — APPROVED** [review]

**Comment by santosomar** [comment]
Discussed and approved during the monthly TC meeting on 2022-02-23.

**Review by santosomar — APPROVED** [review]
Discussed and approved during the monthly TC meeting on 2022-02-23.

**Pull request merged**